samtranslator/model/eventsources/push.py

```diff
@@ -1072,6 +1076,20 @@ def add_auth_to_swagger(  # noqa: PLR0912, PLR0913
             if resource_policy.get("CustomStatements"):
                 editor.add_custom_statements(resource_policy.get("CustomStatements"))  # type: ignore[no-untyped-call]
 
+    @staticmethod
+    def add_timeout_to_swagger(path: str, method_name: str, timeout: int, editor: SwaggerEditor) -> None:
+        """
+        Adds a timeout to this path/method.
+
+        :param path: string of path name
+        :param method_name: string of method name
+        :param timeout: int of timeout duration in milliseconds
+        :param editor: SwaggerEditor object
+
+        """
+        for method_definition in editor.iter_on_method_definitions_for_path_at_method(path, method_name):
+            method_definition[editor._X_APIGW_INTEGRATION]["timeoutInMillis"] = timeout
```

Comment:
Any reason this method is here and not in the `swagger.py` file? Should probably be moved to keep consistency with how http api defines `add_timeout_to_method`

Reply:
I see this is already implemented in `OpenApiEditor`. Is it possible to reuse it by moving it to `BaseEditor`?

Reply:
Right, the two functions are identical and only use BaseEditor constants so I could refactor and move it to `BaseEditor`. Is the use case too specific to be moved there though?

Reply:
We have 2 editor extends `BaseEditor` and both of them require this function. So it makes sense to move there IMO.

Reply:
Updated so `add_timeout_to_method` is now inherited from `BaseEditor`